google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java

```diff
@@ -338,6 +339,90 @@ public String toString() {
     }
   }
 
+  public static class Autoclass implements Serializable {
+
+    private static final long serialVersionUID = -2378172222188072439L;
+    private Boolean enabled;
+    private OffsetDateTime toggleTime;
+
+    public Boolean getEnabled() {
+      return enabled;
+    }
+
+    public OffsetDateTime getToggleTime() {
+      return toggleTime;
+    }
+
+    @Override
+    public boolean equals (Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (!(o instanceof Autoclass)) {
+        return false;
+      }
+      Autoclass that = (Autoclass) o;
+      return Objects.equals(enabled, that.enabled)
+          && Objects.equals(toggleTime, that.toggleTime);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(enabled, toggleTime);
+    }
+
+    @Override
+    public String toString() {
+      return MoreObjects.toStringHelper(this)
+          .add("enabled", enabled)
+          .add("toggleTime", toggleTime)
+          .toString();
+    }
+
+    private Autoclass(Builder builder) {
+      this.enabled = builder.enabled;
+      this.toggleTime = builder.toggleTime;
+    }
+
+    public static Builder newBuilder() {
+      return new Builder();
+    }
+
+    public Builder toBuilder() {
+      return newBuilder()
+          .setEnabled(enabled)
+          .setToggleTime(toggleTime);
+    }
+
+    public static class Builder {
+      private Boolean enabled;
+      private OffsetDateTime toggleTime;
+
```

Comment:
Can we add a private no args constructor to enforce use of `Autoclass#newBuilder()`